src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -164,82 +195,66 @@ static int gavl_intersect_cmp_func(uintptr_t ustart, uintptr_t len, gavl_tree_no
 int MPL_gavl_tree_create(void (*free_fn) (void *), MPL_gavl_tree_t * gavl_tree)
 {
     int mpl_err = MPL_SUCCESS;
-    gavl_tree_s *gavl_tree_iptr;
+    gavl_tree_s *tree_ptr;
 
-    gavl_tree_iptr = (gavl_tree_s *) MPL_malloc(sizeof(gavl_tree_s), MPL_MEM_OTHER);
-    if (gavl_tree_iptr == NULL) {
+    tree_ptr = (gavl_tree_s *) MPL_malloc(sizeof(gavl_tree_s), MPL_MEM_OTHER);
+    if (tree_ptr == NULL) {
         mpl_err = MPL_ERR_SHM_NOMEM;
         goto fn_fail;
     }
```

Comment:
You could use a `MPIR_ERR_CHKANDJUMP` macro, search for '**nomem' for other places that we did the similar.

Reply:
no, this is in MPL, we cannot use `MPIR`

Reply:
You are right. Silly me. In this case, we just need tell the user (MPICH) and MPICH should handle the error properly. Personally I would simply return the opaque `gavl_tree` pointer -- `NULL` indicates error -- but whatever you return here is fine. I think mpich is just checking whether the code is zero anyway.